src/hist.rs

```diff
@@ -0,0 +1,103 @@
+extern crate telemetry;
+
+use self::telemetry::plain::*;
+use std::collections::HashMap;
+
+pub struct Histograms {
+    #[allow(dead_code)] // Issue #33 -- Will go away with subsequent commits.
+    ts: telemetry::Service,
+    hist_storage: HashMap<String, telemetry::plain::Linear<u32>>
+}
+
+impl Histograms {
+    pub fn new() -> Histograms {
+        Histograms {
+            ts: telemetry::Service::new(true /* activate immediately */),
+            hist_storage: HashMap::new()
+        }
+    }
+
+    #[allow(dead_code)] // Issue #33 will go away with subsequent commits.
+    pub fn create_linear_histogram(&mut self, name: &str, min: u32, max: u32, buckets: usize) -> bool  {
+        let hist:telemetry::plain::Linear<u32> = telemetry::plain::Linear::new(
+                    &self.ts,
+                    name.to_string(),
+                    min,
+                    max,
+                    buckets);
+        let ret = self.hist_storage.insert(name.to_string(), hist);
+        match ret {
+            Some(_) => false,
+            None => true
+        }
+    }
+
+    #[allow(dead_code)] // Issue #33 will go away with subsequent commits.
+    pub fn record_linear(&mut self, name: &str, value: u32) -> bool {
+        if self.hist_storage.contains_key(name) {
+            if let Some(hist) = self.hist_storage.get_mut(name) {
+                (*hist).record(value);
+            }
+            return true;
+        } else {
+            return false;
+        }
```

Comment:
You're right.  changed this.  I don't need it here.  You only need it for early returns.  It will complain if I had done an early return like this:
if self.hist_storage.contains_key(name {
  true
}
false